### PR TITLE
Implemented the Auth module, login and logout components and cleaned up app.component

### LINKED ISSUE #153: Refreshing on any page in the dashboard redirects user to the home page.
### Description
When a user navigates to any page in the dashboard and refreshes the browser window using the browser refresh button the user is redirected to the home page.

### What should happen?
If a user refreshes the browser the dashboard should remember the context and stay in the same page.

### What actually happens?
The user is redirected to the home page.

The likely cause for this is not maintaining the state properly in the router.  

### Developers Guide to setup Development environment
The developer guide to setup a development environment can be found [here](https://docs.sodafoundation.io/guides/developer-guides/dashboard/)

### LINKED ISSUE #650: Create an authentication module and move all auth related code to this module.
**Issue/Feature Description:**
Create an auth module and amove all the auth related code from app.controller.ts to appropriate controllers.

- [ ] Implement Auth Module
- [ ] Implement login component
- [ ] Implement logout component
- [ ] Implement auth guard

**Why this issue to fixed / feature is needed(give scenarios or use cases):**
Currently the authentication is taken care using the Keystone service. The login and logout scripts are written in the app controller and there is no separate module.
To ensure that the the dashboard authentication code is segregated from the app component and  can be used with any other auth service we develop the auth module.

**How to reproduce, in case of a bug:**

**Other Notes /  Environment Information: (Please give the env information, log link or any useful information for this issue)**

### PR DESCRIPTION
**Pull request by ravisaxena23**
**What type of PR is this?**
 /kind new feature
 /kind bug fix
 /kind cleanup
 /kind enhancement

**What this PR does / why we need it**:
We have optimized the login process. added the login code to the new auth module, and cleaned the app component.
This PR also fixes the issue present on #153. this is because we have moved the login code to a different module so now the router does not navigate to home every time.
This PR  Implements Auth Module, login component, and logout component.

**Which issue(s) this PR fixes**:
Fixes #650 
Fixes #153

**Test Report Added?**:
TESTED
I have tested the login and logout and some scenarios as well.  I am sorry but I am unable to test consts file changes currently please help me with it.

**Test Report**
Screenshot (7)
Screenshot (8)
Screenshot (9)
Screenshot (10)

**Special notes for your reviewer**:
I have tested the login and logout and some scenarios as well.  I am sorry but I am unable to test consts file changes currently please help me with it. 
I am sorry my linter has made small changes.

### CODE REVIEW

**Review by anvithks — CHANGES_REQUESTED** [review]
Some initial observations. Could you PTAL

File: src/app/business/auth/login/login.component.ts [hunk 1]
```diff
@@ -0,0 +1,356 @@
...
+                // Check if the user is admin and assign to a window variable
+                window['isAdmin'] = (_.where(roles, { 'name': "admin" })).length > 0;
+
+                // check if the user is a regular user of type Member and assign to window variable
+                window['isUser'] = (_.where(roles, { 'name': "Member" })).length > 0;
+
+                this.paramStor.AUTH_TOKEN(r.headers.get('x-subject-token'));
+                this.paramStor.CURRENT_TENANT(project.name + "|" + project.id);
+                this.paramStor.CURRENT_USER(user.name + "|" + user.id);
+                this.paramStor.CURRENT_ROLE(JSON.stringify(roles))
```
[the remark this answers is not in the document]
Comment (ravisaxena23) [reply src/app/business/auth/login/login.component.ts]: remove this line this.paramStor.CURRENT_ROLE(JSON.stringify(roles))

File: src/app/business/auth/logout/logout.component.html [hunk 2]
```diff
@@ -0,0 +1,3 @@
+<h1>
+  logout works!
```
[the remark this answers is not in the document]
Comment (ravisaxena23) [reply src/app/business/auth/logout/logout.component.html]: removed dummy files from both login and logout

**Comment by ravisaxena23** [comment]
@anvithks  I have added a blank logout template as the template is required for the component. it was showing an error on the console If the file is not present. further, I have moved menu_items to the login component, if it is required we can create a separate issue for it.

**Comment by anvithks** [comment]
Some more testing is needed to verify if any other pages are affected. 
But overall it looks like the Auth module has been implemented properly. 
Next we add the Auth guard to the routes.

**Comment by ravisaxena23** [comment]
Fixes #153

**Review by joseph-v — APPROVED** [review]
LGTM

**Review by anvithks — APPROVED** [review]
LGTM

**Pull request merged**